src/components/common/view/Image.js

```diff
@@ -28,24 +29,19 @@ const Image = ({ style = {}, onLoad = noop, ...props }) => {
         }
   }, [flattenStyle, aspectRatio])
 
-  const onImageLoaded = useCallback(
-    event => {
-      const [getStyle, onLoad] = refs
+  useEffect(() => {
+    const onImageSize = (width, height) => {
+      const [getStyle] = refs
 
       if (isAutoHeight(getStyle())) {
-        const dimensions = getDimensions(event.nativeEvent)
-
-        if (dimensions) {
-          setAspectRatio(dimensions.width / dimensions.height)
-        }
+        setAspectRatio(width / height)
       }
+    }
 
-      onLoad(event)
-    },
-    [setAspectRatio, refs],
-  )
+    NativeImage.getSize(uri, onImageSize, e => log.error('Get image size error', e.message, e))
```

Comment:
why do we need this ? doesn't it makes duplicated request ?

Reply:
i've added image displaying only after we'll got ratio
hope this prevents extra req

Reply:
Yes, it works even without [setAspectRatio, refs] dependancies, checked web/android/ios